torch_xla/stablehlo.py

```diff
@@ -266,17 +271,25 @@ def run_node(self, n) -> Any:
         dynamic_dims = [
             i for i, x in enumerate(fake_t.shape) if not isinstance(x, int)
         ]
-        self._mark_dynamic(res, dynamic_dims)
       return res
-    if n.op == 'call_function':
+    elif n.op == 'call_function':
       if hasattr(n.target, 'namespace'
                 ) and n.target.namespace in self.custom_ops_allowed_in_graph:
         output_shapes, output_dtypes = extract_custom_call_outputs_shape_dtype(
             n)
         call_name = str(n.target)
         n.target = stablehlo_custom_call
         n.args = (n.args, call_name, output_shapes, output_dtypes)
-    return super().run_node(n)
+      res = super().run_node(n)
+    else:
+      res = super().run_node(n)
+    if self.gm_serializer is not None:
+      node_metadata = json.dumps(self.gm_serializer.serialize_metadata(n))
+      pytree.tree_map_only(
+          torch.Tensor,
+          lambda x: torch_xla._XLAC._set_xla_custom_op_name_prefix(
```

Comment:
Why not just use `torch_xla.experimental.xla_mlir_debuginfo.write_mlir_debuginfo`? https://github.com/pytorch/xla/blob/40f7e1f54b506475d40b40c0f49193411de6d68f/torch_xla/experimental/xla_mlir_debuginfo.py#L17

Reply:
Yeah, I think it's a good idea, it also inserts the marker which the MLIR pass would look at. Updated!

Reply:
Just saw Han's comment. https://github.com/pytorch/xla/blob/40f7e1f54b506475d40b40c0f49193411de6d68f/torch_xla/experimental/xla_mlir_debuginfo.py#L10 enables `XLA_HLO_DEBUG` when import. You may want to remove it. I will do corresponding changes in ai-edge-torch.